USAGE.md

```diff
@@ -3,34 +3,80 @@
+CONTAINER_RUNTIME=podman
+VAULT_LICENSE=02MV4UU..
+echo CONTAINER_RUNTIME=${CONTAINER_RUNTIME:-docker} > .env
+if [[ -z ${CONTAINER_RUNTIME##docker} ]];then echo CONTAINER_DOMAIN=docker.internal; else echo CONTAINER_DOMAIN=dns.podman; fi >> .env
+echo VAULT_LICENSE=${VAULT_LICENSE} >> .env
+echo IPA_REALM=${CONTAINER_DOMAIN^^} >> .env
+```
+
+## 3. Start Containers
 
 Start FreeIPA, Keycloak, Vault and K3s:
 ```bash
-docker-compose up -d
+sudo ${CONTAINER_RUNTIME}-compose up -d
 ```
 
-## 3. Setup Resolution of Domain Names
+## 4. Setup Resolution of Domain Names
+
+You can either add a static record for every container to your `/etc/hosts` file or configure systemd-resolved service with `resolvectl` to use the container environment internal DNS (recommended).
+
+### Option 1: Configuration using container internal dns (recommended)
+First, find out which interface is used
+```bash
+CONTAINER_RUNTIME_IF=$(sudo ${CONTAINER_RUNTIME} network inspect vault-playground_vault-playground | jq -j '.[].network_interface')
+```
+which IP it has assigned
+```bash
+CONTAINER_RUNTIME_IF_IP=$(ip -j -p -br -4 addr show dev ${CONTAINER_RUNTIME_IF} | jq -j '.[].addr_info | .[].local')
+```
+and then configure this interface to resolve the domain specified
+```bash
+sudo -E resolvectl dns ${CONTAINER_RUNTIME_IF} ${CONTAINER_RUNTIME_IF_IP}
+sudo -E resolvectl domain ${CONTAINER_RUNTIME_IF} ${CONTAINER_DOMAIN}
+```
 
+### Option 2: Static configuration using hosts file
 Add a static A record for the container hosting FreeIPA, Keycloak and Vault. Alternatively, adjust your `/etc/hosts` file accordingly:
 
 ```bash
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} ipa.identity.net{{end}}' ipa) | sudo tee -a /etc/hosts
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} keycloak.identity.net{{end}}' keycloak) | sudo tee -a /etc/hosts
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} vault.identity.net{{end}}' vault) | sudo tee -a /etc/hosts
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} k3s-server.identity.net{{end}}' k3s-server) | sudo tee -a /etc/hosts
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} grafana.identity.net{{end}}' grafana) | sudo tee -a /etc/hosts
-echo -e $(docker inspect -f '{{range .NetworkSettings.Networks}}{{.IPAddress}} prometheus.identity.net{{end}}' prometheus) | sudo tee -a /etc/hosts
+echo -e $(sudo -E ${CONTAINER_RUNTIME} inspect -f "{{range .NetworkSettings.Networks}}{{.IPAddress}} ipa.${CONTAINER_DOMAIN}{{end}}" ipa) | sudo tee -a /etc/hosts
+echo -e $(sudo -E ${CONTAINER_RUNTIME} inspect -f "{{range .NetworkSettings.Networks}}{{.IPAddress}} keycloak.${CONTAINER_DOMAIN}{{end}}" keycloak) | sudo tee -a /etc/hosts
+echo -e $(sudo -E ${CONTAINER_RUNTIME} inspect -f "{{range .NetworkSettings.Networks}}{{.IPAddress}} vault.${CONTAINER_DOMAIN}{{end}}" vault) | sudo tee -a /etc/hosts
```

Comment:
Vault and k3s don't have an IP on my podamn setup

Reply:
I suspect this is because k3s and Vault are not running, right? please verify and resolve this if you agree on that

Reply:
yes let's assume this @Xelef2000, right? I'll go ahead with this one :fast_forward:

Reply:
@Xelef2000 I encountered the same today. If you naively just hit `docker-compose up -d` without first properly setting all the required environment variables (especially, `CONTAINER_RUNTIME`, `CONTAINER_DOMAIN`, `IPA_REALM`), then you end up in the situation where the k3s and thee ipa container will not start. So, solution is to setup the `env` file as mentioned in the usage instructions or export the above mentioned variables. I hop this helps.